rasa/core/policies/embedding_policy.py

```diff
@@ -71,215 +42,168 @@ class EmbeddingPolicy(Policy):
-        bot_embed: Optional[tf.Tensor] = None,
-        slot_embed: Optional[tf.Tensor] = None,
-        dial_embed: Optional[tf.Tensor] = None,
-        rnn_embed: Optional[tf.Tensor] = None,
-        attn_embed: Optional[tf.Tensor] = None,
-        copy_attn_debug: Optional[tf.Tensor] = None,
-        all_time_masks: Optional[tf.Tensor] = None,
+        graph: Optional["tf.Graph"] = None,
+        session: Optional["tf.Session"] = None,
+        user_placeholder: Optional["tf.Tensor"] = None,
+        bot_placeholder: Optional["tf.Tensor"] = None,
+        similarity_all: Optional["tf.Tensor"] = None,
+        pred_confidence: Optional["tf.Tensor"] = None,
+        similarity: Optional["tf.Tensor"] = None,
+        dial_embed: Optional["tf.Tensor"] = None,
+        bot_embed: Optional["tf.Tensor"] = None,
+        all_bot_embed: Optional["tf.Tensor"] = None,
+        attention_weights: Optional["tf.Tensor"] = None,
+        max_history: Optional[int] = None,
         **kwargs: Any
     ) -> None:
-        if featurizer:
-            if not isinstance(featurizer, FullDialogueTrackerFeaturizer):
-                raise TypeError(
-                    "Passed tracker featurizer of type {}, "
-                    "should be FullDialogueTrackerFeaturizer."
-                    "".format(type(featurizer).__name__)
-                )
-        super(EmbeddingPolicy, self).__init__(featurizer, priority)
+        """Declare instant variables with default values"""
 
-        # flag if to use the same embeddings for user and bot
-        try:
-            self.share_embedding = self.featurizer.state_featurizer.use_shared_vocab
-        except AttributeError:
-            self.share_embedding = False
+        if not featurizer:
+            featurizer = self._standard_featurizer(max_history)
+        super(EmbeddingPolicy, self).__init__(featurizer, priority)
 
         self._load_params(**kwargs)
 
-        # chrono initialization for forget bias
-        self.characteristic_time = None
-
-        # encode all actions with numbers
-        # persist this array for prediction time
-        self.encoded_all_actions = encoded_all_actions
+        # encode all labels with numbers
+        self._encoded_all_labels = None
 
         # tf related instances
         self.graph = graph
         self.session = session
-        self.a_in = intent_placeholder
-        self.b_in = action_placeholder
-        self.c_in = slots_placeholder
-        self.b_prev_in = prev_act_placeholder
-        self._dialogue_len = dialogue_len
-        self._x_for_no_intent_in = x_for_no_intent
-        self._y_for_no_action_in = y_for_no_action
-        self._y_for_action_listen_in = y_for_action_listen
-        self.sim_op = similarity_op
-
-        # store attention probability distribution as
-        # concatenated tensor of each attention types
-        self.alignment_history = alignment_history
+        self.a_in = user_placeholder
```

Comment:
`a_in` and `b_in` are not really self explainable names. Can we rename them to something more meaningful?

Reply:
the problem is that they are used in both nlu classifier and core policy. Where they are different things, but algorithm-wise the same. @dakshvar22 any suggestions?